main/ahb_for_sles.go

```diff
@@ -97,61 +98,88 @@ func parseCfg(filename string) (map[string]map[string]string, error) {
 	return ini, nil
 }
 
-func _isRegistered() (bool, error) {
-	services, err := filepath.Glob("/etc/zypp/services.d/*.service")
-	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
-		return false, err
+func _getSUSEConnectStatus() (bool, bool, error) {
+	commandOutput, error := RunShellCommand(0, "SUSEConnect", "-s")
+	if error != nil {
+		fmt.Println(error)
+		return false, false, error
+	}
+	var suseConnectStatus []map[string]interface{}
+	json.Unmarshal([]byte(commandOutput), &suseConnectStatus)
+	subscriptionStatus := fmt.Sprintf("%v", suseConnectStatus[0]["subscription_status"])
+	status := fmt.Sprintf("%v", suseConnectStatus[0]["status"])
+	if strings.ToLower(status) == "registered" && strings.ToLower(subscriptionStatus) == "active" {
+		// system is registered with an active subscription
+		// check if services are present
+		services, err := filepath.Glob("/etc/zypp/services.d/*.service")
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return true, true, err
+		}
+		if len(services) == 0 {
+			extensionsOutput, err := RunShellCommand(0, "SUSEConnect", "--list-extensions")
+			if err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				return true, true, err
+			}
+			extensionsList := strings.Split(extensionsOutput, "\n")
+			for _, extension := range extensionsList {
+				if strings.Contains(extension, "Deactivate with") {
+					// activate whatever it was active AND
+					// Public Cloud module
+					start := strings.Index(string(extension), "SUSEConnect")
+					if start != -1 {
+						command := string(extension)[start:len(extension)]
+						commandList := strings.Split(command, " ")
+						_, err = RunShellCommand(0, commandList[0], commandList[2], commandList[3])
+						if err != nil {
+							fmt.Fprintln(os.Stderr, err)
+							return true, true, err
+						}
+					}
+				}
+			}
+			return true, true, err
+		}
+		// check if repos are present
+		repos, err := filepath.Glob("/etc/zypp/repos.d/*.repo")
```

Comment:
If we have services then `zypper` will set up the repos. For example you can `rm -rf /etc/zypp/repos.d/*.repo` on a system and as long as the service is set up can run `zypper up` and the repos will be restored. I am not certain why we need the extra check for repositories. I think conceptually there is some confusion about this use case implementation.